packages/usa-combo-box/src/index.js

```diff
@@ -382,28 +382,63 @@ const displayList = (el) => {
   const filter = comboBoxEl.dataset.filter || DEFAULT_FILTER;
   const regex = generateDynamicRegExp(filter, inputValue, comboBoxEl.dataset);
 
-  const options = [];
-  for (let i = 0, len = selectEl.options.length; i < len; i += 1) {
-    const optionEl = selectEl.options[i];
-    const optionId = `${listOptionBaseId}${options.length}`;
-
-    if (
-      optionEl.value &&
-      (disableFiltering ||
-        isPristine ||
-        !inputValue ||
-        regex.test(optionEl.text))
-    ) {
-      if (selectEl.value && optionEl.value === selectEl.value) {
-        selectedItemId = optionId;
+  let options = [];
+  const optionsAlphabetical = [];
+  const optionsContains = [];
+  const optionList = [...selectEl.options];
+
+  /**
+   * Builds options array.
+   *
+   * If disableFiltering or a selection has been made: do not filter.
```

Comment:
Currently, an empty input is used to reset the filter. If we do an early return, the filter would persist after the last character is backspaced.

Since the display is built per event, I would recommend leaving for now unless we decide to do a larger refactor.

Reply:
@mahoneycm my use case was for initial default state where user clicks on combo box, the list is generated and shown.

If I user simply toggles it without selecting a value, the entire list is recreated.

Reply:
This should be addressed in #6193 (More detail in [this comment](https://github.com/uswds/uswds/pull/6122#issuecomment-2465721547)).